### ISSUE TITLE
can't convert vbox network to terraform data: No match with get guestproperty output

### ISSUE THREAD

**Issue by git006** [terra-farm/terraform-provider-virtualbox#138]
resource "virtualbox_vm" "node" { 
        count = 1 
        name = "${format("k8s-%02d", count.index+1)}" 
        url = "https://cloud.centos.org/centos/7/vagrant/x86_64/images/CentOS-7-x86_64-Vagrant-2004_01.VirtualBox.box"
        image = "./VirtualBox.box"
        cpus = 1
        memory = "512 Mib"
        network_adapter { 
            type = "bridged"
            host_interface = "wlp0s20f3"
           # device = "wlp0s20f3"
        }
} 

I installed virtualbox in my ubuntu.  What value should i provide to the host_interface or device? i just check ifconfig  and got the adapter name

**Comment by fabdg** [comment]
I have the same problem using https://github.com/ccll/terraform-provider-virtualbox-images/releases/download/ubuntu-15.04/ubuntu-15.04.tar.xz image

**Comment by PTangeL** [comment]
same here, I think the VM it's still starting after 60 seconds, there is a PR that put a timeout as a variable, so we could tweak it easily.

**Comment by Jean-Baptiste-Lasselle** [comment]
Hello, quick msg to give a full explanation of how to successfully use the provider, testd with ubuntu and debian, with full explanations and details about the caveats and limitations of the provider: https://github.com/3forges/packman/tree/master/terraform

**Comment by CyraxNova** [comment]
@Jean-Baptiste-Lasselle I add PR (https://github.com/terra-farm/go-virtualbox/pull/45) for fix bridge-adapter.

@git006 VirtualBox gold image need contain VBox GuestAdditons (https://www.oracle.com/virtualization/technologies/vm/downloads/virtualbox-downloads.html)

For more detail see
`VBoxManage guestproperty enumerate "vm-name"`

**Comment by Jean-Baptiste-Lasselle** [comment]
> @Jean-Baptiste-Lasselle I add PR ([terra-farm/go-virtualbox#45](https://github.com/terra-farm/go-virtualbox/pull/45)) for fix bridge-adapter.
> 
> @git006 VirtualBox gold image need contain VBox GuestAdditons (https://www.oracle.com/virtualization/technologies/vm/downloads/virtualbox-downloads.html)
> 
> For more detail see `VBoxManage guestproperty enumerate "vm-name"`



:) Oh, someone read [my long analysis ](https://github.com/3forges/packman/blob/master/terraform/README.md) :) (Ctrl + F `where the option syntax is now --bridge-adapter<1-N>, and not --bridgeadapter<1-N> anymore. I hereafter, checked that it still` in that analysis of mine) ... I see, and thank you for pinging me I'm happy this could help, I could have done the PR, but I have other plans actually, to bring up a different design for terraforming virutalbox VMs, which is why I was interested in the study :) ... And As I have explained it, the virtualbox provider from `terra-farm` still works even if it will break soon. But it's true a PR will make the provider last longer.

All in all, it still is a good surprise, that there still exists this one out of ten thousand, who would read a very long, in-depth analysis :) ... And thank you for the mention :)

**Comment by CyraxNova** [comment]
@Jean-Baptiste-Lasselle : Yes. I'm using `image     = "https://app.vagrantup.com/ubuntu/boxes/jammy64/versions/20240315.1.0/providers/virtualbox.box"` (Ubuntu 22.04). By default, the image does not contain VirtualBox guest extensions. Need to install VirtualBox guest extensions on the vm and repackage the image.  Not `VM VirtualBox Extension Pack`. Exactly `VBox GuestAdditons`.

**Comment by Jean-Baptiste-Lasselle** [comment]
> @Jean-Baptiste-Lasselle : Yes. I'm using `image = "https://app.vagrantup.com/ubuntu/boxes/jammy64/versions/20240315.1.0/providers/virtualbox.box"` (Ubuntu 22.04). By default, the image does not contain VirtualBox GuestAdditons. Need to install VirtualBox GuestAdditons on the vm and repackage the image. Not `VM VirtualBox Extension Pack`. Exactly `VBox GuestAdditons`.

Ahhhh Thank you so very much, I will test that as soon as possible: I like it very much when every single detail is completely analyzed, and in my case, i made the terraform provider work, without the error message anymore, simply because I commented the `device` configuration option. So I will add to my study this detail, appreciate very much your indication, which i will definitely test and add to my study :)

**Comment by CyraxNova** [comment]
@Jean-Baptiste-Lasselle VBox GuestAdditons need for work all func in https://github.com/terra-farm/go-virtualbox/blob/main/guestprop.go
